src/stringSimilarity.js

```diff
@@ -1,17 +1,103 @@
 import escapeRegExp from 'lodash.escaperegexp';
 import matcher from 'matcher';
+import { isObject } from './helpers';
 
 export const WILDCARD_MARKER_ESCAPED = '{{\\*}}';
 export const WILDCARD_MARKER = '{{*}}';
+export const UNQUOTED_WILDCARD_PLACEHOLDER = '{{%}}';
+const searchUnquotedWildcards = new RegExp(/(([^:\s,]*)\{\{\*\}\}([^,:\s}\]]*))(?=[^,:}\]\s]*)/g);
+const searchModifiedWildcards = new RegExp(/(([^:\s,]*)\{\{%\}\}([^,\s}\]]*))(?=[^,}\]\s]*)/g);
 
-export default (source, target) => {
-  if (!source || (source || '') === (target || '')) {
-    return source === target;
-  }
-
+const getIsMatch = (source, target) => {
   const wildcardedSource = source
     .replace(new RegExp(escapeRegExp('*'), 'g'), '\\*')
     .replace(new RegExp(escapeRegExp(WILDCARD_MARKER_ESCAPED), 'g'), '*');
 
   return matcher.isMatch(target, wildcardedSource);
 };
+
+export const recursiveKeySort = (data) => {
+  if (data && isObject(data)) {
+    const sortedData = {};
+    Object.keys(data).sort().forEach((key) => {
+      sortedData[key] = recursiveKeySort(data[key]);
+    });
+    if (Array.isArray(data)) {
+      return Object.values(sortedData);
+    }
+    return sortedData;
+  }
+  return data;
+};
+
```

Comment:
While I hate duplicate code, I wonder if in this case if it would be worth a few lines of duplicate code if we broke the function `replaceWildcards` into two functions: `replaceWildcards` and `replaceWildcardPlaceholders`? That might make it clearer what is happening in the main code when `replaceWildcards` is called twice. Your thoughts?

Reply:
Agreed and done, let me know what you think!

Reply:
I went with `restoreWildcards` rather than `replaceWildcardPlaceholders`, hope that works for you as well.